### DISCUSSION TITLE
Reading a .Epub file from Raspberry Pi Pico using Micropython

### DISCUSSION THREAD

**Discussion by kumaaarrr** [micropython/micropython#12673] [General]
Hello,
I've been currently wanting to read a .epub file from a SD card using Pi Pico but I'm not able to succeed. I tried a few python libraries from the internet but no luck. Is there any library by Micropython for reading a .epub file as text or is there any other way for it?

Thanks.

**Comment by scruss** [comment]
This may be more difficult than it looks. While there might be libraries to open Zip files (epub files are special cases of Zip files) all the control metadata in an EPUB is in XML. There really isn't a good XML parser for MicroPython

**Reply by kumaaarrr** [reply]
Hey thanks for replying, So is there any other way like can we convert the file format and then read it. My application is that I want to read an epub file from an SD card and display the content.

**Comment by fdufnews** [comment]
There is an open ebook reader project by Joey Castillo using a Pico but is not using MicroPython.
https://www.oddlyspecificobjects.com/projects/openbook/
Maybe you can analyze the code and port it to MicroPython or change your mind and use the original C software.
